autoblocks/_impl/configs/config.py

```diff
@@ -0,0 +1,246 @@
+            )
+
+        global_state.init()
+        task: AnyTask
+        if running_loop := get_running_loop():
+            # If we're already in a running loop, execute the task on that loop
+            task = running_loop.create_task(
+                self._load_and_set_remote_config(
+                    config=config, api_key=api_key, timeout=activate_timeout, parser=parser
+                ),
+            )
+        else:
+            # Otherwise, send the task to our background loop
+            task = asyncio.run_coroutine_threadsafe(
+                self._load_and_set_remote_config(
+                    config=config, api_key=api_key, timeout=activate_timeout, parser=parser
+                ),
+                global_state.event_loop(),
+            )
+
+        # Wait for config to be initialized
+        task.result()
+
+        # we do not refresh the config if we are in a testing context
+        if is_remote_config_refreshable(config) and not is_testing_context():
+            if running_loop := get_running_loop():
+                running_loop.create_task(
+                    self._refresh_loop(
+                        config=config,
+                        api_key=api_key,
+                        refresh_interval=refresh_interval,
+                        refresh_timeout=refresh_timeout,
+                        parser=parser,
+                    )
+                )
+            else:
+                asyncio.run_coroutine_threadsafe(
+                    self._refresh_loop(
+                        config=config,
+                        api_key=api_key,
+                        refresh_interval=refresh_interval,
+                        refresh_timeout=refresh_timeout,
+                        parser=parser,
+                    ),
+                    global_state.event_loop(),
+                )
+
+    def activate_from_remote(
+        self,
+        config: RemoteConfig,
+        parser: Callable[[Dict[str, Any]], AutoblocksConfigValue],
+        api_key: Optional[str] = None,
+        refresh_interval: timedelta = timedelta(seconds=10),
+        refresh_timeout: timedelta = timedelta(seconds=30),
+        activate_timeout: timedelta = timedelta(seconds=30),
+    ) -> None:
+        """
+        Activate a remote config from Autoblocks and optionally refresh it every `refresh_interval` seconds.
+        """
+        try:
+            log.info("Activating remote config...")
+            self._activate_from_remote_unsafe(
+                config=config,
+                api_key=api_key,
+                refresh_interval=refresh_interval,
+                refresh_timeout=refresh_timeout,
+                activate_timeout=activate_timeout,
+                parser=parser,
+            )
+        except Exception as err:
+            log.error(f"Failed to activate remote config '{config.id}': {err}")
+
+    @property
+    def value(self) -> AutoblocksConfigValue:
```

Comment:
do we need to put access behind a context manager so we can reliably track what version was used during a request?